app/models/checkout.rb

```diff
@@ -1,34 +1,43 @@
 class Checkout < ApplicationRecord
   CHECKOUT_PERIOD_IN_DAYS = 14
-  FINE_AMOUNT_PER_DAY_IN_CENTS = 0.10
+  FINE_AMOUNT_PER_DAY_IN_CENTS = 10
 
   belongs_to :user
   belongs_to :book
 
   validates :user, presence: true
-  validates :book, presence: true, uniqueness: true
+  validates :book, presence: true #, uniqueness: true
+  validates :due_on, presence: true
 
-  def due_date
-    self.created_at + CHECKOUT_PERIOD_IN_DAYS.days
-  end
+  monetize :fine_cents
 
-  def fine
-    number_of_days_overdue = days_overdue
+  def calculate_fine
     if overdue?
-      fine = FINE_AMOUNT_PER_DAY_IN_CENTS * days_overdue
+      number_of_days_overdue = days_overdue
+      checkout_fine = fine_cents + number_of_days_overdue * FINE_AMOUNT_PER_DAY_IN_CENTS
     else
-      fine = 0
+      checkout_fine = fine_cents
     end
+    Money.new(checkout_fine)
+  end
+
+  def calculate_fine_including_related_checkouts
+    book_checkouts = Checkout.where(user_id: self.user_id, book_id: self.book_id)
+    total = book_checkouts.inject(Money.new(0)) { |total, book_checkout| total + book_checkout.calculate_fine }
+    total
   end
 
   def overdue?
-    number_of_days_overdue = days_overdue
-    number_of_days_overdue > 0 ? true : false
+    if closed_at.present? # checkout is not overdue if it has been closed
+      false
+    else
+      days_overdue > 0 ? true : false
+    end
   end
 
   private
 
   def days_overdue
-    (Time.zone.now - due_date).to_i/1.day
+    (Time.zone.now - self.due_on).to_i/1.day
```

Comment:
I don't think you need `self` for this. Also it's a bit hard to see the division here. I'd put spaces around the `/`